internal/provisioner/plain/plain.go

```diff
@@ -0,0 +1,102 @@
+package plain
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/operator-framework/operator-controller/internal/bundledeployment"
+	"github.com/operator-framework/operator-controller/internal/util"
+)
+
+const (
+	// ProvisionerID is the unique plain provisioner ID
+	ProvisionerID = "core-rukpak-io-plain"
+
+	manifestsDir = "manifests"
+)
+
+func HandleBundleDeployment(_ context.Context, fsys fs.FS, _ *bundledeployment.BundleDeployment) (*chart.Chart, chartutil.Values, error) {
```

Comment:
Right; and that brings up another larger issue for this PR: Rukpak was kind of infamous for testing with e2e tests instead of unit tests, so we can't bring over a lot of those tests as-is. Personally I think there's probably too much new code in this PR for adding unit tests for all of it and I'd prefer to open a new issue for that if you think that's reasonable.

Reply:
sounds good.to me. just need an issue for adding rukpakian related unit test, to do later

Reply:
Created #1037